contracts/examples/Moon.sol

```diff
@@ -0,0 +1,255 @@
+     */
+    function triggerChannelClose(bytes32 channelId) external onlyOwner {
+        dispatcher.channelCloseInit(channelId);
+    }
+
+    /**
+     * @dev Sends a packet with a greeting message over a specified channel.
+     * @param message The greeting message to be sent.
+     * @param channelId The ID of the channel to send the packet to.
+     * @param timeoutTimestamp The timestamp at which the packet will expire if not received.
+     */
+    function greet(string calldata message, bytes32 channelId, uint64 timeoutTimestamp) external {
+        dispatcher.sendPacket(channelId, bytes(message), timeoutTimestamp);
+    }
+
+    function onChanOpenInit(ChannelOrder, string[] calldata, string calldata, string calldata version)
+        external
+        view
+        virtual
+        onlyIbcDispatcher
+        returns (string memory selectedVersion)
+    {
+        return _openChannel(version);
+    }
+
+    // solhint-disable-next-line ordering
+    function onChanOpenTry(
+        ChannelOrder,
+        string[] memory,
+        bytes32 channelId,
+        string memory,
+        bytes32,
+        string calldata counterpartyVersion
+    ) external virtual onlyIbcDispatcher returns (string memory selectedVersion) {
+        return _connectChannel(channelId, counterpartyVersion);
+    }
+
+    function onChanOpenAck(bytes32 channelId, bytes32, string calldata counterpartyVersion)
+        external
+        virtual
+        onlyIbcDispatcher
+    {
+        _connectChannel(channelId, counterpartyVersion);
+    }
+
+    function onChanOpenConfirm(bytes32 channelId) external onlyIbcDispatcher {}
+
+    function _connectChannel(bytes32 channelId, string calldata counterpartyVersion)
+        private
+        returns (string memory version)
+    {
+        // ensure negotiated version is supported
+        for (uint256 i = 0; i < supportedVersions.length; i++) {
+            if (keccak256(abi.encodePacked(counterpartyVersion)) == keccak256(abi.encodePacked(supportedVersions[i]))) {
+                connectedChannels.push(channelId);
+                return counterpartyVersion;
+            }
+        }
+        revert UnsupportedVersion();
+    }
+
+    function _openChannel(string calldata version) private view returns (string memory selectedVersion) {
+        for (uint256 i = 0; i < supportedVersions.length; i++) {
+            if (keccak256(abi.encodePacked(version)) == keccak256(abi.encodePacked(supportedVersions[i]))) {
+                return version;
+            }
+        }
+        revert UnsupportedVersion();
+    }
+
+    function onChanCloseInit(bytes32 channelId, string calldata, bytes32) external virtual onlyIbcDispatcher {}
+}
+
+/*
+ * These contracts are the exact same as Moon, but they revert/panick in different ways.
+ * they are used to test that transport error is seperated from errors thrown in onRecvPacket
+ */
+contract RevertingStringMoon is Moon {
```

Comment:
I think we can get rid of all the revertingString contracts since those are only used in unit test suite. You won't ever need to deploy these to testnets even for load testing